### PR TITLE
Restrict Field Mutability Based on Status

### LINKED ISSUE #23: Restrict Field Mutability Based on Status
Currently, the fields **title**, **description**, **category**, and **priority** are editable even after the status of an item transitions to **started** or **done**. This introduces a logical flaw, as these fields should ideally remain immutable once work has commenced.  

### Expected Behavior  
1. Restrict the editing of **title**, **description**, **category**, and **priority** fields once the status becomes **started** or **done**.  
2. Allow these fields to remain editable only when the status is set to **not started**.  

### Tasks:  
1. **Logic Implementation:**  
   - Update the field validation logic in the `frontend` (`EditTicketForm.jsx`) as well as `backend` (`PUT` function in the `/api/[id]/route.js`) to enforce immutability based on status.  
   - Ensure attempts to edit restricted fields result in appropriate feedback or an error message.  

2. **UI Updates:**  
   - Disable the respective fields in the UI when the status is not **not started**.  
   - Provide tooltips or notifications explaining why the fields are locked.

### Acceptance Criteria:  
- **Title**, **description**, **category**, and **priority** are editable only when the status is **not started**.  
- Fields are appropriately disabled in the UI based on the item's status.  
- Attempts to edit locked fields produce meaningful feedback.

### PR DESCRIPTION
**Pull request by disvid**
### Related Issue  
Closes #23   

### Type of Change
Put `x` inside the square bracket to specify what type of change your PR is:  
- [ ] New Feature  
- [x] Bug Fix  
- [ ] Code Refactor  
- [ ] Documentation Update  
- [ ] Other (please specify): 

### Description of Change  
Restricts the editing of title, description, category, and priority fields once the status becomes started or done.
Allows these fields to remain editable only when the status is set to not started.

### Implementation Details  
required changes have been made in app/(components)/EditTicketForm.jsx and app/api/Tickets/route.js these files.

### Demo  
[Screencast from 2024-12-21 05-03-00.webm](https://github.com/user-attachments/assets/090676d5-eb4b-4982-854a-f990f36b88dc)

### CODE REVIEW

**Comment by AyushDharDubey** [comment]
Hi @disvid,  

great work on the frontend part! however, I believe there’s been a slight misinterpretation regarding the API endpoint for updating tickets.  

for updating a ticket, we send a PUT request to `/api/Tickets/[id]`, **not** `/api/Tickets/`. a PUT function for this purpose is already defined in [`route.js`](https://github.com/IMGIITRoorkee/Ticketify/blob/master/app/api/Tickets/%5Bid%5D/route.js).  

please avoid defining a new PUT function in [`/api/Tickets/route.js`](https://github.com/IMGIITRoorkee/Ticketify/blob/master/app/api/Tickets/route.js). instead, use and update the one already defined at [`/api/Tickets/[id]/route.js`](https://github.com/IMGIITRoorkee/Ticketify/blob/master/app/api/Tickets/%5Bid%5D/route.js). 

thank you

**Comment by disvid** [comment]
@AyushDharDubey Done sir, the changes you mentioned have been incorporated.

**Comment by AyushDharDubey** [comment]
Please resolve the merge conflict

**Comment by disvid** [comment]
@AyushDharDubey Done sir.

File: app/api/Tickets/[id]/route.js [hunk 1]
```diff
@@ -14,10 +14,22 @@ export async function PUT(req, { params }) {
 
     const body = await req.json();
     const ticketData = body.formData;
+    const ticketId = req.url.split("/").pop();
```
Comment (AyushDharDubey) [inline app/api/Tickets/[id]/route.js:17]: we are already extracting the ticketId as a parameter of PUT function on line no 13

```javascript
export async function PUT(req, { params }) {
  try {
    const { id } = params;

    const body = await req.json();
```

so no need for this new ticketId object

Comment (disvid) [reply app/api/Tickets/[id]/route.js]: Done sir, made the required change.

Comment (AyushDharDubey) [reply app/api/Tickets/[id]/route.js]: did you test the project locally after your latest commit?

it appears you removed the `ticketId` declaration in your latest commit but didn’t update the reference where it’s being called.


```javascript
const existingTicket = await Ticket.findById(ticketId);
```

 
 line no 17

Comment (disvid) [reply app/api/Tickets/[id]/route.js]: Sorry sir, made the changes.

**Comment by AyushDharDubey** [comment]
Hi @disvid,

good work on this one, i can affirm that the fix is working and is ready to be merged
thank you for your contribution

**Pull request merged**